enhancements/20250820-support-multipath.md

```diff
@@ -0,0 +1,285 @@
+
+4. **Installer Configuration Extension:** Extended Harvester installer to accept additional multipath configuration parameters including WWID-based identification or `blacklist_exception` for better usage. For example, below configuraiton is used for testing. We can use `wwid "!^0QEMU_QEMU_HARDDISK_disk[0-9]+"` in `blacklist` section or `wwid "^0QEMU_QEMU_HARDDISK_disk[0-9]+"` in `blacklist_exceptions`. Both are okay. However, I think it's more flexible to provide the different configuration for users.
+    ```
+    blacklist {
+        device {
+            vendor "!QEMU"
+            product "!QEMU HARDDISK"
+        }
+        wwid ".*"
+    }
+    blacklist_exceptions {
+        wwid "^0QEMU_QEMU_HARDDISK_disk[0-9]+"
+    }
+    ```
+
+### Test plan
+
+We should prepare two different clusters to test:
+
+1. Cluster with multipath service enabled by default.
+2. Cluster with multipath service disabled by default.
+
+For the second cluster, after the cluster is up, we need to run `systemctl enable multipathd` and `systemctl start multipathd`. After that, continue with the following test steps.
+
+Additionally, you need to prepare two LUNs pointing to the same disk. If you're not sure about the setup, it should look like this in the `virt-install` command line:
+
+```
+# Same serial numbers, and same physical disk
+--disk path=/var/lib/libvirt/images/mydisk1.img,bus=scsi,cache=none,shareable=on,serial=disk1 \
+--disk path=/var/lib/libvirt/images/mydisk1.img,bus=scsi,cache=none,shareable=on,serial=disk1 \
+```
+
+Then, start testing:
+
+1. In the settings, enable Longhorn V2 engine
+2. Run `multipath -l` to verify that the multipath service is working properly
+3. Add the specific multipath disk (for Longhorn V1) and another regular disk (for Longhorn V2) to Harvester through the GUI
+4. Create two VM images using Longhorn V1 and V2 engines respectively
+5. Start two virtual machines with the images and additional disks:
+    - Longhorn V1 Image with Longhorn V1 extra disk
+    - Longhorn V2 Image with Longhorn V2 extra disk
+6. After starting the VMs, reboot the node. Verify that the virtual machines come back up successfully after the reboot
+
+### Upgrade strategy
+
+Some users might have already enabled multipathd and are using it. In such cases, the disks should be configured and added in the Longhorn UI by the users themselves. 
+
+If those disks are available in the Harvester UI, adding them in the Harvester UI will cause failure. Therefore, NDM or GUI should filter out disks that are already mounted and added in the Longhorn UI.
+
+By the way, if you add a normal disk and configure it in the Longhorn UI, it's still available in the Harvester UI. Adding it also causes failure. So, the behavior has been like this for a long time. The suitable way to migrate it from Longhorn to Harvester is to delete replica and backing images before adding the disks on the Harvester side.
```

Comment:
is there a separate issue tracking this behaviour?

Reply:
I haven't opened it. Let me open it later.

Reply:
Will the failure to add a wrongly recognized device cause data loss? Need to consider data protection if needed.

Reply:
I just tested it again. It seems if there are replicas, we still can schedule replicas on the disk even disk is added in the Harvester UI. It just show "target is busy".